src/it/scala/io/iohk/ethereum/txExecTest/util/DumpChainActor.scala

```diff
@@ -111,8 +111,8 @@ class DumpChainActor(peerManager: ActorRef, peerMessageBus: ActorRef, startBlock
       val nodes = NodeData(stateNodes).values.indices.map(i => NodeData(stateNodes).getMptNode(i))
 
       val children = nodes.flatMap {
-        case n: BranchNode => n.children.collect { case HashNode(h) => h }
-        case ExtensionNode(_, HashNode(h), _, _) => Seq(h)
+        case n: BranchNode => n.children.collect { case HashNode(h) => ByteString(h) }
```

Comment:
When we are dealing with hashes we are doing a lot of boxing/unboxing or making a conversions. Probably is not the issue for now but maybe we should think about using one type everywhere

Reply:
For now i focused to avoid as much conversions as possible inside mpt trie, and left those outside intact. If benchmarks will show that that they are somehow bottleneck then it can be tackled in other pr.